--- Utilities/cmlibuv/src/unix/fs.c	2019-02-03 10:48:16.000000000 -0600
+++ Utilities/cmlibuv/src/unix/fs.c	2019-02-03 13:12:52.000000000 -0600
@@ -125,6 +125,12 @@
   }                                                                           \
   while (0)

+#if defined(__MirBSD__)
+# define CANCEL_ERROR -ENOCOFFEE
+#else
+# define CANCEL_ERROR -ECANCELED
+#endif
+

 static ssize_t uv__fs_fdatasync(uv_fs_t* req) {
 #if defined(__linux__) || defined(__sun) || defined(__NetBSD__)
@@ -810,6 +816,13 @@
   dst->st_birthtim.tv_nsec = src->st_ctimensec;
   dst->st_flags = 0;
   dst->st_gen = 0;
+#elif defined(__MirBSD__)
+  dst->st_atim.tv_sec = src->st_atime;
+  dst->st_atim.tv_nsec = src->st_atimensec;
+  dst->st_mtim.tv_sec = src->st_mtime;
+  dst->st_mtim.tv_nsec = src->st_mtimensec;
+  dst->st_ctim.tv_sec = src->st_ctime;
+  dst->st_ctim.tv_nsec = src->st_ctimensec;
 #elif !defined(_AIX) && (       \
     defined(__DragonFly__)   || \
     defined(__FreeBSD__)     || \
@@ -1004,9 +1017,9 @@
   req = container_of(w, uv_fs_t, work_req);
   uv__req_unregister(req->loop, req);

-  if (status == -ECANCELED) {
+  if (status == CANCEL_ERROR) {
     assert(req->result == 0);
-    req->result = -ECANCELED;
+    req->result = CANCEL_ERROR;
   }

   req->cb(req);
