--- Utilities/cmlibuv/src/unix/fs.c	2019-02-03 10:48:16.000000000 -0600
+++ Utilities/cmlibuv/src/unix/fs.c	2019-02-03 12:39:07.000000000 -0600
@@ -810,6 +810,13 @@
   dst->st_birthtim.tv_nsec = src->st_ctimensec;
   dst->st_flags = 0;
   dst->st_gen = 0;
+#elif defined(__MirBSD__)
+  dst->st_atim.tv_sec = src->st_atime;
+  dst->st_atim.tv_nsec = src->st_atimensec;
+  dst->st_mtim.tv_sec = src->st_mtime;
+  dst->st_mtim.tv_nsec = src->st_mtimensec;
+  dst->st_ctim.tv_sec = src->st_ctime;
+  dst->st_ctim.tv_nsec = src->st_ctimensec;
 #elif !defined(_AIX) && (       \
     defined(__DragonFly__)   || \
     defined(__FreeBSD__)     || \
@@ -1004,10 +1011,12 @@
   req = container_of(w, uv_fs_t, work_req);
   uv__req_unregister(req->loop, req);

+  #if !defined(__MirBSD__)
   if (status == -ECANCELED) {
     assert(req->result == 0);
     req->result = -ECANCELED;
   }
+  #endif

   req->cb(req);
 }
