--- Utilities/cmlibuv/src/unix/getnameinfo.c	2017-07-18 10:22:56.000000000 -0500
+++ Utilities/cmlibuv/src/unix/getnameinfo.c	2019-02-03 12:55:48.000000000 -0600
@@ -61,10 +61,13 @@
   uv__req_unregister(req->loop, req);
   host = service = NULL;

+  #if !defined(__MirBSD__)
   if (status == -ECANCELED) {
     assert(req->retcode == 0);
     req->retcode = UV_EAI_CANCELED;
-  } else if (req->retcode == 0) {
+  } else
+  #endif
+  if (req->retcode == 0) {
     host = req->host;
     service = req->service;
   }
