--- Utilities/cmlibuv/src/unix/thread.c	2017-07-18 10:22:56.000000000 -0500
+++ Utilities/cmlibuv/src/unix/thread.c	2019-02-03 15:20:04.000000000 -0600
@@ -424,7 +424,7 @@
   if (err)
     return -err;

-#if !(defined(__ANDROID__) && defined(HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC))
+#if !defined(__MirBSD__) && !(defined(__ANDROID__) && defined(HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC))
   err = pthread_condattr_setclock(&attr, CLOCK_MONOTONIC);
   if (err)
     goto error2;
